storage/invoke.go

```diff
@@ -27,6 +27,25 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+// run determines whether a retry is necessary based on the config and
+// idempotency information. It then calls the function with or without retries
+// as appropriate, using the configured settings.
+func run(ctx context.Context, call func() error, retry *retryConfig, isIdempotent bool) error {
+	if !isIdempotent {
+		return call()
+	}
+	bo := gax.Backoff{}
+	if retry != nil && retry.backoff != nil {
+		bo.Multiplier = retry.backoff.Multiplier
+		bo.Initial = retry.backoff.Initial
+		bo.Max = retry.backoff.Max
```

Comment:
Why are we setting each field individually rather than the object as a whole?

Reply:
We need to copy/clone the backoff rather than using the provided one, since the actual struct itself is stateful and can't be reused for multiple calls. See https://github.com/googleapis/gax-go/blob/v2.1.1/v2/call_option.go#L129